### PR TITLE
Fix null value in flakey test search.

### PR DESCRIPTION
**Pull request by yusuf-goog**
Bug:104099

*Replace this paragraph with a description of what this PR is changing or adding, and why. Consider including before/after screenshots.*

*List which issues are fixed by this PR. You must list at least one issue.*

*If you had to change anything in the [flutter/tests] repo, include a link to the migration guide as per the [breaking change policy].*

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read the [Flutter Style Guide] _recently_, and have followed its advice.
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by yusuf-goog** [comment]
Fixes https://github.com/flutter/flutter/issues/104099

File: app_dart/lib/src/request_handlers/check_flaky_builders.dart [hunk 1]
```diff
@@ -163,8 +164,9 @@ class CheckFlakyBuilders extends ApiRequestHandler<Body> {
   }
 
   bool _getIgnoreFlakiness(String? builderName, CiYaml ciYaml) {
-    final Target target = ciYaml.postsubmitTargets.singleWhere((Target target) => target.value.name == builderName);
-    return target.getIgnoreFlakiness();
+    final Target? target =
```
Comment (keyonghan) [inline app_dart/lib/src/request_handlers/check_flaky_builders.dart:167]: Can we add a test to cover this?

Comment (yusuf-goog) [reply app_dart/lib/src/request_handlers/check_flaky_builders.dart:167]: Do you have advice on a easy way to get to this function or spoof the data to trigger the issue? Need to have builderName be a value thats not found in the ciYaml object.

Comment (keyonghan) [reply app_dart/lib/src/request_handlers/check_flaky_builders.dart:167]: an easy way is to make this function visible from test, and test this function specifically.

Comment (yusuf-goog) [reply app_dart/lib/src/request_handlers/check_flaky_builders.dart:168]: added tests

**Review by keyonghan — COMMENTED** [review]
LGTM

**Review by XilaiZhang — APPROVED** [review]

**Review by keyonghan — APPROVED** [review]
Oops, thought I have approved..

**Pull request merged**